tests/rptest/services/redpanda.py

```diff
@@ -146,11 +146,14 @@ def __str__(self):
             return f"<NodeCrash ({names}) {example}>"
 
 
+V = TypeVar('V', int, float)
+
+
 class MetricSamples:
-    def __init__(self, samples):
+    def __init__(self, samples: List[MetricSample]):
         self.samples = samples
 
-    def label_filter(self, labels):
+    def label_filter(self, labels: Mapping[str, V]):
```

Comment:
Two ways to express a type constraint. I'm cool with either way. Typevar can be very concise, which is nice, but I think explicit inline types might be clearer here.

Reply:
fine with either but if we are using TypeVar maybe we should use a more expressive name, especially since this is in the most included file in the tests tree?

Reply:
I just discovered `numbers.Number`.. I'll replace the typevar with this type.

Reply:
Tried Number and Real ..  but they don't quite work. FWIW I guess the recommendation is just to use `float`. [link](https://peps.python.org/pep-0484/#the-numeric-tower)

Reply:
You can use `Union[int, float]]`?